### DISCUSSION TITLE
Does not work with RHvoice on ubuntu 18.04.5

### DISCUSSION THREAD

**Discussion by edvejak** [johnfactotum/foliate#754] [Q&A]
I can hear the app in settings (with spd-say -e -w -p 50). I chose RHvoice voice synthesizer. I open cyrillic text (utf-8) and I hear nothing. The app don't speak at all
Terminal error: 
`$ /usr/bin/com.github.johnfactotum.Foliate
Gjs-Message: 01:39:23.592: JS WARNING: [resource:///com/github/johnfactotum/Foliate/js/window.js 289]: reference to undefined property "direction"
Gjs-Message: 01:39:23.592: JS WARNING: [resource:///com/github/johnfactotum/Foliate/js/window.js 152]: reference to undefined property "direction"
Gjs-Message: 01:39:23.765: JS WARNING: [resource:///com/github/johnfactotum/Foliate/js/window.js 313]: reference to undefined property "layout"

(com.github.johnfactotum.Foliate:3292): Foliate-WARNING **: 01:39:48.523: TypeError: lang is undefined

(com.github.johnfactotum.Foliate:3292): Foliate-WARNING **: 01:39:57.105: TypeError: lang is undefined

(com.github.johnfactotum.Foliate:3292): Foliate-WARNING **: 01:40:37.966: TypeError: lang is undefined

(com.github.johnfactotum.Foliate:3292): Foliate-WARNING **: 01:40:42.326: TypeError: lang is undefined

(com.github.johnfactotum.Foliate:3292): Foliate-WARNING **: 01:40:44.133: TypeError: lang is undefined

(com.github.johnfactotum.Foliate:3292): Gjs-WARNING **: 01:40:46.698: Unhandled promise rejection. To suppress this warning, add an error handler to your promise chain with .catch() or a try-catch block around your await expression. Stack trace of the failed promise:
  set epub/<@resource:///com/github/johnfactotum/Foliate/js/window.js:165:13
  _handleAction@resource:///com/github/johnfactotum/Foliate/js/epubView.js:920:17
  _init/<@resource:///com/github/johnfactotum/Foliate/js/epubView.js:617:13
  main@resource:///com/github/johnfactotum/Foliate/js/main.js:476:12
  run@resource:///org/gnome/gjs/modules/package.js:225:12
  @/usr/bin/com.github.johnfactotum.Foliate:9:1`
I've installed deb package release com.github.johnfactotum.foliate_2.6.3_all.deb

**To Reproduce**
1. Open Foliate
2. Choose file.txt
3. press F5
4. The app don't speak

**Expected behavior**
The app must pronounce the chosen text

**Version:**
 - Foliate version:  2.6.3
 - OS/Distribution and version: Ubuntu 18.04.5
 - Desktop environment: GNOME Shell 3.28.4
 - Installation method: deb package

**Comment by archisman-panigrahi** [comment]
Is the app working otherwise, except the voice synthesizer part?

Are other voice synthesizers like espeak working?

**Comment by edvejak** [comment]
The app opens text file. I can read text in the app window. I use RHvoice voice synthesizer and it works in firefox for example. Otherwise I can hear the app in settings (with spd-say -e -w -p 50). Espeak isn't installed, I don't use it.

**Comment by johnfactotum** [comment]
@vantu5z seemed to have used RHVoice successfully; see https://github.com/johnfactotum/foliate/issues/481#issuecomment-647380483.

**Comment by vantu5z** [comment]
@edvejak, It works for me with `spd-say -e -w -p 50` settings.
For myself I use additional scripts to run Rhvoice as mentioned in #481 (additional info https://github.com/vantu5z/RHVoice-dictionary/issues/213#issuecomment-644757312).
Do you try set Foliate settings like this: `RHVoice-test -p Artemiy`?
